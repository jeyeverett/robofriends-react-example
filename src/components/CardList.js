import React from 'react';
import Card from './Card';

//Note that default parameter is 'props' (we can call it whatever we want but it is the general object containing all of the props defined in the JSX) which means we have to destructure it in the param field (or use props.robots notation everywhere)
const CardList = ({ robots }) => {
    //Below we create an array of Card elements by cycling through the robots array and storing their values in he JSX as shown. Note that we need to provide a 'key' for each Card element so that React can uniquely identify each one (we will get a warning if we omit it)
    //Ultimately we don't need to store the result of robots.map in a variable, we can put the code directly inside the div below inside {}
    // const cardComponent = robots.map((user, i) => {
    //     return (
    //         <Card 
    //             key={i} 
    //             id={robots[i].id} 
    //             name={robots[i].name} 
    //             email={robots[i].email}
    //         />);
    // })

    return (
        <div>
            {
               robots.map((user, i) => {
                return (
                    <Card 
                        key={i} 
                        id={robots[i].id} 
                        name={robots[i].name} 
                        email={robots[i].email}
                    />);
                })
            }
        </div>
    );
}

export default CardList;