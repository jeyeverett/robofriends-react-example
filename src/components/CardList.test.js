import { create } from "react-test-renderer";
import React from 'react';
import CardList from './CardList';

const mockRobots = [
    {
        id: 1,
        name: 'John Snow',
        username: 'Jon',
        email: 'john@gmail.com'
    }
];

const cardList = create(<CardList robots={mockRobots} />);

it('expect to render CardList component', () => {
    expect(cardList.toJSON()).toMatchSnapshot()
});