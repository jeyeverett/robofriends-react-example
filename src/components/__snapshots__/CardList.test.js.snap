// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div>
  <div
    className="bg-green dib br3 pa4 ma2 grow bw2 shadow-5 tc"
  >
    <img
      alt="RoBro"
      height="200"
      src="https://robohash.org/1?size=200x200&set=set3"
      width="200"
    />
    <div>
      <h2
        className="ttu white f4"
      >
        John Snow
      </h2>
      <p
        className="ttu white i"
      >
        john@gmail.com
      </p>
    </div>
  </div>
</div>
`;
